src/Roslyn.Diagnostics.Analyzers/CSharp/CSharpAvoidOptSuffixForNullableEnableCode.cs

```diff
@@ -66,7 +66,14 @@ private static void ReportOnInvalidIdentifier(SyntaxToken identifier, SemanticMo
             }
 
             var symbol = semanticModel.GetDeclaredSymbol(identifier.Parent, cancellationToken);
-            if (symbol?.GetMemberOrLocalOrParameterType()?.NullableAnnotation() == Analyzer.Utilities.Lightup.NullableAnnotation.Annotated)
+            if (symbol?.GetMemberOrLocalOrParameterType()?.NullableAnnotation() != Analyzer.Utilities.Lightup.NullableAnnotation.Annotated)
+            {
+                return;
+            }
+
+            if (symbol.Kind != SymbolKind.Parameter ||
+                symbol.ContainingSymbol == null ||
+                (!symbol.ContainingSymbol.IsOverride && !symbol.ContainingSymbol.IsImplementationOfAnyInterfaceMember()))
```

Comment:
Nit: At this point it seems it would be more readable to write this with a static local function `ShouldReport(ISymbol)` with explicit return false for parameters whose containing member is an implementaiton.